tendrl/ceph_integration/sds_sync/__init__.py

```diff
@@ -67,17 +67,20 @@ def __init__(self):
     def _ping_cluster(self):
         NS.tendrl_context = NS.tendrl_context.load()
         NS.node_context = NS.node_context.load()
-        if NS.tendrl_context.cluster_id:
+        try:
+            if NS.tendrl_context.cluster_id:
                 cluster_data = ceph.heartbeat(NS.tendrl_context.cluster_id)
-                NS.tendrl_context.cluster_id = self.fsid = cluster_data['fsid']
-        else:
-            cluster_data = ceph.heartbeat()
-            if cluster_data:
+            else:
+                cluster_data = ceph.heartbeat()
+            if None not in cluster_data:
                 if "fsid" in cluster_data:
                     NS.tendrl_context.cluster_id = self.fsid = \
                         cluster_data['fsid']
-
-        NS.tendrl_context.cluster_name = self.name = cluster_data['name']
+                NS.tendrl_context.cluster_name = self.name = cluster_data['name']
+        except Exception as ex:
```

Comment:
I see this `_ping_cluster` being called while initialization of the SDS sync thread so pass should be fine as using this we just populate the cluster-id and cluster-name fields. The only thing is some of the details down the line might fail to sync but they would be logged with ERROR and that's correct I feel.

If there are errors while syncing the cluster data user anyway need to check the state/status of the cluster and debug further why that's happening.

@r0h4n ^^ ??